flink-streaming-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/Kafka09Fetcher.java

```diff
@@ -289,7 +289,11 @@ public void commitSpecificOffsetsToKafka(Map<KafkaTopicPartition, Long> offsets)
 		Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>(partitions.length);
 
 		for (KafkaTopicPartitionState<TopicPartition> partition : partitions) {
-			Long offset = offsets.get(partition.getKafkaTopicPartition());
+			/*
+			 * Increment offset by one, otherwise last record will be read again. This does not affect checkpoints/saved state.
+			 * The offset is only read from Kafka/ZK on a fresh startup of a job, not restart or failure. See https://issues.apache.org/jira/browse/FLINK-4618
```

Comment:
nit: I think its ok to change the link to simply refer to the Kafka 0.9 API instead of the JIRA, because this is the proper use of the {{commitAsync}} method.